🎁 Hide empty collection metadata

kirkkwang opened the pull request:
This commit will bring over two partials from Hyrax so that we can hide empty metadata fields in the collection show page and the dashboard. The reason why they show in the first place is because Bulkrax brings them in as arrays with empty strings.

Ref:
  - https://github.com/scientist-softserv/utk-hyku/issues/235

# Expected Behavior Before Changes

Empty metadata fields is showing up in the collection's metadata

# Expected Behavior After Changes

There should be no more empty metadata fields.

# Screenshots / Video

<details><summary>Before</summary>

![image](https://github.com/scientist-softserv/utk-hyku/assets/19597776/72bbb7ab-aa6e-4894-be07-c589ba2b03fd)

![image](https://github.com/scientist-softserv/utk-hyku/assets/19597776/a8002835-e4a7-494f-80e8-289c332db8fc)

</details>

<details><summary>After</summary>

![image](https://github.com/scientist-softserv/utk-hyku/assets/19597776/1d757242-570e-4e13-9bf9-304883619974)

![image](https://github.com/scientist-softserv/utk-hyku/assets/19597776/0ff9f8c9-8b78-4102-bb3b-7a7c3bc2acfc)

</details>

alishaevn reviewed (changes requested):
did you try overriding [#terms_with_values](https://github.com/samvera/hyrax/blob/main/app/presenters/hyrax/collection_presenter.rb#L50-L52) to do the right thing?
instead of overriding the views and creating `#render_metadata_field` to essentially do the same thing that the current views are already doing?

kirkkwang commented:
Ohh nice, i'll give that a shot! thanks for the tip

Commit:
🎁 Hide empty collection metadata

This commit will override the #terms_with_values method in the
CollectionPresenter class to hide empty metadata fields in the
collection show page.  The reason why they show in the first place is
because Bulkrax brings them in as arrays with empty strings.

Ref:
  - https://github.com/scientist-softserv/utk-hyku/issues/235
